src/control/KinematicSimulationTrajectoryExecutor.cpp

```diff
@@ -77,8 +79,13 @@ std::future<void> KinematicSimulationTrajectoryExecutor::execute(
     mPromise.reset(new std::promise<void>());
 
     mTraj = traj;
-    mStateSpace = std::dynamic_pointer_cast<MetaSkeletonStateSpace>(
-        traj->getStateSpace());
+
+    auto nonConstStateSpace
+        = std::const_pointer_cast<aikido::statespace::StateSpace>(
+            traj->getStateSpace());
+    mStateSpace
+        = std::dynamic_pointer_cast<MetaSkeletonStateSpace>(nonConstStateSpace);
```

Comment:
I believe we could make `mStateSpace` const so that `const_pointer_cast` is not required.

Reply:
I wanted to get rid of `const_pointer_cast` as well but we initialize the executor dependent only on the skeleton (mStateSpace{nullptr}) and tie it to a statespace` only after the trajectory is received in `execute`. 
1. We could pass the statespace when the executor is instantiated.
2. We could get rid of `mStateSpace` entirely but this would mean extracting it out of the trajectory everytime `step` is called.

Reply:
I think we could achieve this by changing `traj` and `mTraj` to const. I believe a trajectory executor doesn't need to change the trajectory so it would make sense to take trajectory as const.

Reply:
I thought an executor for a skeleton can `execute()` on multiple trajectories.

Reply:
It would be fine. You can take a new _const_ trajectory and set it to `mTraj`, which is also a const trajectory. Please note that the constness applies to the object but not the pointer itself so we can change the pointer to another pointer (but can't change the object the pointer points to; a trajectory executor executes multiple trajectories but doesn't change the parameters of the trajectories). Let me know if I'm missing something here.

Reply:
OOOOH! I got confused again! Dammit. :disappointed:
Yeah that makes sense.

Reply:
Done. :)
Didn't have to change const-ness of the trajectories but `mTraj` and `traj` should actually be pointers to const objects as well. I will address that while I work on #282